src/desktop.c

```diff
@@ -170,42 +139,30 @@ desktop_cycle_view(struct server *server, struct view *start_view,
 	/* Make sure to have all nodes in their actual ordering */
 	osd_preview_restore(server);
 
-	if (!start_view) {
-		start_view = first_view(server);
-		if (!start_view || start_view != server->active_view) {
-			return start_view;  /* may be NULL */
-		}
-	}
```

Comment:
With the removal of this I think we break the the 2nd point about the XWayland views (subview with focus but not on top of the list) in the comment above.

Reply:
Good point.

Not sure how to best deal with that. 

We could consider a `criteria` for toplevel-with-no-parent.

Reply:
I've added a root-toplevel criteria, so the problem has gone away.

Reply:
> I've added a root-toplevel criteria, so the problem has gone away.

Didn't test yet but doesn't this break the ability of selecting sub windows (like `Open file` / `Search` / `About` dialogues)?

Reply:
Yes, intentionally so.

`desktop_focus_view()` calls `view_move_to_front()` which moves all sub-views (e.g. modal dialogs) with it, so selecting the root-toplevel moves the `Open File` / `Search` / `About` with it. 

Having been wanting to make this change for a long time, but it wasn't a sensible change prior to this `osd.c` / `desktop_cycle_view()` refactoring.

I haven't looked at what other OS/DE/WM/Compositors do, but suspect they just show root toplevels in the window-switchers. I might test a couple of other tomorrow.

Reply:
Well, not all subwindows are modal dialogues. Usually `Search` in a GUI editor is one of those.
We would then not have the ability to switch between the `Search` window and the main window via `A-Tab` anymore.

Reply:
In firefox, manage bookmarks brings up a window, when a selection is made 
the main window gets focus, and the bookmark window is now hidden.
To get to it (without alt-tab) would require me to either move the main window enough to see the 
one behind or possibly minimize the main window. Neither option is great.

Several apps have the behavior of subwindows being there until manually closed.

Reply:
Well, we could just apply the root-toplevel when pre-selecting the second window, but please consider the below first:

1. If a "subwindow" (dialog or whatever) is a separate root-toplevel, then it still shows up in the window-switcher (this happens with gimp, inkscape's About window, etc.)
2. If a subwindow is a indeed a child-view then it moves to back/front with its root-toplevel so I struggle to see the benefit of listing it separately in the window-switcher.

Note: "subwindow" used loosely here as you might argue the first case doesn't apply

Reply:
To me there are 2 types of "sub" windows, 
dialogs like open, library, etc
drop down like menu selection (ff, etc) but they usually go away with change of focus.

Waybar's taskbar shows 2 windows for firefox when I have library open for example.
but if I click on a menu then I don't see a second window in taskbar.

I assume there's some way to tell the difference between the 2 types, and maybe what you're
doing might work.

Reply:
> 2\. If a subwindow is a indeed a child-view then it moves to back/front with its root-toplevel so I struggle to see the benefit of listing it separately in the window-switcher.

For me it's about fast keyboard focus switching without having to use the mouse.

Reply:
Did a quick test, using root toplevel, seems to work ok.

I started ff, and selected manage bookmarks, was able to alt-tab between both windows
and they show up in the osd, firefox has whatever title from main window, bookmarks shows
library as title. 

Not sure how Xwayland windows will behave though, as I don't have that installed.